examples/wildcard-https.yaml

```diff
@@ -0,0 +1,27 @@
+kind: Gateway
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: gateway
+  namespace: default
+spec:
+  class: default-class
+  addresses:
+  - type: NamedAddress
+    value: auto-assign
+  listeners:
+  - hostname:
+      match: Domain
+      value: example.com
```

Comment:
Should this be `*.example.com`?

Reply:
It's not intended to be. For host name `foo.example.com` the domain is `example.com`. I assumed that requiring the leading `*.` didn't make things clearer :)